$NetBSD: patch-ad,v 1.20 2016/08/16 23:53:25 ryoon Exp $

*BSD and qnx are identified as linux.
Add another SDK path on Mac OS X.
All sparc cpus can not do unaligned access.
Detect NetBSD ARMv7 hardfloat toolchain.

--- build/make/configure.sh.orig	2016-07-21 01:15:41.000000000 +0000
+++ build/make/configure.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/pkg/bin/bash
 ##
 ##  configure.sh
 ##
@@ -683,7 +683,7 @@ process_common_toolchain() {
       armv6*)
         tgt_isa=armv6
         ;;
-      armv7*-hardfloat* | armv7*-gnueabihf | arm-*-gnueabihf)
+      armv7*-hardfloat* | armv7*-gnueabihf | arm-*-gnueabihf | armv7*-netbsd*-*hf)
         tgt_isa=armv7
         float_abi=hard
         ;;
@@ -735,7 +735,7 @@ process_common_toolchain() {
         [ -z "$tgt_isa" ] && tgt_isa=x86
         tgt_os=win32
         ;;
-      *linux*|*bsd*)
+      *linux*|*bsd*|*qnx6*)
         tgt_os=linux
         ;;
       *solaris2.10)
@@ -778,6 +778,9 @@ process_common_toolchain() {
     mips*)
       enable_feature mips
       ;;
+    sparc*)
+      disable_feature fast_unaligned
+      ;;
   esac
 
   # PIC is probably what we want when building shared libs
@@ -1415,7 +1418,7 @@ EOF
   check_cc <<EOF
 unsigned int e = 'O'<<24 | '2'<<16 | 'B'<<8 | 'E';
 EOF
-    [ -f "${TMP_O}" ] && od -A n -t x1 "${TMP_O}" | tr -d '\n' |
+    [ -f "${TMP_O}" ] && od -t x1 "${TMP_O}" | tr -d '\n' |
         grep '4f *32 *42 *45' >/dev/null 2>&1 && enable_feature big_endian
 
     # Try to find which inline keywords are supported
@@ -1431,11 +1434,33 @@ EOF
       *-android-gcc)
         ;;
       *)
-        check_header pthread.h && add_extralibs -lpthread
+        check_header pthread.h && add_extralibs ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}
         ;;
     esac
   fi
 
+  case ${tgt_os} in
+  darwin*)
+     add_extralibs -lm
+     ;;
+  solaris*)
+     add_extralibs -lm -lrt
+     ;;
+  linux*)
+     case ${gcctarget} in
+     *qnx6*)
+         add_extralibs -lm
+         ;;
+     *openbsd*)
+         add_extralibs -lm
+         ;;
+     *)
+         add_extralibs -lm -lrt
+         ;;
+     esac
+     ;;
+  esac
+
   # only for MIPS platforms
   case ${toolchain} in
     mips*)
